src/Tutorials/GausianProcessRegressor.cs

```diff
@@ -0,0 +1,150 @@
+
+            for (int i = 0; i < trainingInputs.Length; i++)
+            {
+                Gaussian post = sgp.Marginal(trainingInputs[i]);
+                double postMean = post.GetMean();
+                var xTrain = trainingInputs[i][0];
+                meanSeries.Points.Add(new DataPoint(xTrain, postMean));
+                scatterSeries.Points.Add(new OxyPlot.Series.ScatterPoint(xTrain, trainingOutputs[i]));
+
+                var mean = 0.0;
+                var precision = 0.0;
+                post.GetMeanAndPrecision(out mean, out precision);
+                var stdDev = System.Math.Sqrt(1 / precision);
+                areaSeries.Points.Add(new DataPoint(xTrain, postMean + (2 * stdDev)));
+                areaSeries.Points2.Add(new DataPoint(xTrain, postMean - (2 * stdDev)));
+            }
+
+            var model = new PlotModel();
+            model.Title = "Gaussian Process trained on Auto Insurance in Sweden dataset";
+            model.Series.Add(meanSeries);
+            model.Series.Add(scatterSeries);
+            model.Series.Add(areaSeries);
+            model.Axes.Add(new OxyPlot.Axes.LinearAxis {
+                Position = OxyPlot.Axes.AxisPosition.Bottom,
+                Title = "x (number of claims)" });
+            model.Axes.Add(new OxyPlot.Axes.LinearAxis {
+                Position = OxyPlot.Axes.AxisPosition.Left,
+                Title = "y (total payment for all the claims)" });
+
+            // Required for plotting
+            Thread thread = new Thread(() => PngExporter.Export(model, OutputPlotPath, 600, 400, OxyColors.White));
+            thread.SetApartmentState(ApartmentState.STA);
+            thread.Start();
+
+            Console.WriteLine("Saved PNG to {0}", OutputPlotPath);
+        }
+#endif
+
+        private IEnumerable<(double x, double y)> LoadAISDataset()
+        {
+            var data = new List<(double x, double y)>();
+
+            // Read CSV file
+            using (var reader = new StreamReader(AisCsvPath))
+            {
+                while (!reader.EndOfStream)
+                {
+                    var line = reader.ReadLine();
+                    var values = line.Split(',');
+                    if (values.Length == 2)
+                    {
+                        data.Add((double.Parse(values[0]), double.Parse(values[1])));
+                    }
+                }
+            }
+
+            return PreprocessData(data).ToList();
+        }
+
+        private IEnumerable<(double x, double y)> PreprocessData(
+            IEnumerable<(double x, double y)> data)
+        {
+            var x = data.Select(tup => tup.x);
+            var y = data.Select(tup => tup.y);
+
+            // Shift targets so mean is 0
+            var meanY = y.Sum() / y.Count();
+            y = y.Select(val => val - meanY);
+
+            // Scale data to lie between 1 and -1
+            var absoluteMaxY = y.Select(val => System.Math.Abs(val)).Max();
```

Comment:
Better to choose a more robust statistic; variance scaling is slightly more robust for example. A single max/min being an outlier will cause this approach to squash say 99% of the data into a small range and leave the 1% remaining data at the extreme -1/1

Reply:
Thanks mate I'll do that instead